packages/eslint-plugin/tests/rules/no-useless-template-literals.test.ts

```diff
@@ -0,0 +1,240 @@
+import { noFormat, RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-useless-template-literals';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootPath = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: rootPath,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('no-useless-template-literals', rule, {
+  valid: [
+    "const string = 'a';",
+
+    `
+      const string = 'a';
+      const concatenated = \`\${string}b\`;
+    `,
+
+    `
```

Comment:
Question: How are we handling conflicts with [`quotes`](https://typescript-eslint.io/rules/quotes/)? Been on a couple projects where the backtick is preferred, and it would be nice if we could scope this to only strings _with_ template literals if possible.

Should add tests for these imo:

```ts
// For example, these should be valid (even if they flag other rules)
const a = `A`; 
const b = `$B`;
const c = '${"C"}';
const d = "${'D'}";
const e = '${`C`}';
const f = "${`D`}";
```

Reply:
https://github.com/typescript-eslint/typescript-eslint/pull/7957#discussion_r1416275932 - yeah this rule shouldn't try to enforce style preferences. It should only target unnecessary logic.

Reply:
Fixed, thanks!